lib/components/os/resinos.js

```diff
@@ -71,3 +67,28 @@ exports.injectNetworkConfiguration = (image, configuration) => {
     path: '/system-connections/resin-wifi'
   }, wifiConfiguration.join('\n'))
 }
+
+module.exports = class ResinOS {
+  constructor (image, fetch, configuration) {
+    this.image = image
+    this.fetch = fetch
```

Comment:
This looks very weird. The `ResinOS` OS component should surely know how to fetch itself, so I think this logic should live here.